src/main/java/edu/illinois/cs/dt/tools/detection/DetectorPlugin.java

```diff
@@ -187,92 +187,124 @@ public static int moduleRounds(String coordinates) throws IOException {
 
     @Override
-    public void execute(final MavenProject mavenProject) {
-        final ErrorLogger logger = new ErrorLogger(mavenProject);
+    public void execute(final ProjectWrapper project) {
+        final ErrorLogger logger = new ErrorLogger(project);
         this.coordinates = logger.coordinates();
 
-        logger.runAndLogError(() -> detectorExecute(logger, mavenProject, moduleRounds(coordinates)));
+        logger.runAndLogError(() -> detectorExecute(logger, project, moduleRounds(coordinates)));
     }
 
-    private Void detectorExecute(final ErrorLogger logger, final MavenProject mavenProject, final int rounds) throws IOException {
+    private Void detectorExecute(final ErrorLogger logger, final ProjectWrapper project, final int rounds) throws IOException {
         Files.deleteIfExists(DetectorPathManager.errorPath());
         Files.createDirectories(DetectorPathManager.cachePath());
         Files.createDirectories(DetectorPathManager.detectionResults());
 
-        final Option<Runner> runnerOption = RunnerFactory.from(mavenProject);
-
-        // We need to do two checks to make sure that we can run this project
-        // Firstly, we must be able to run it's tests (if we get a runner from the RunnerFactory, we're good)
-        // Secondly, there must be some tests (see below)
-        if (runnerOption.isDefined()) {
-            if (this.runner == null) {
-                this.runner = InstrumentingSmartRunner.fromRunner(runnerOption.get());
-            }
-
-            final List<String> tests = getOriginalOrder(mavenProject);
-
-            // If there are no tests, we can't run a flaky test detector
-            if (!tests.isEmpty()) {
-                Files.createDirectories(outputPath);
-                Files.write(DetectorPathManager.originalOrderPath(), String.join(System.lineSeparator(), tests).getBytes());
-
-                final Detector detector = DetectorFactory.makeDetector(runner, tests, rounds);
-                TestPluginPlugin.info("Created dependent test detector (" + detector.getClass() + ").");
-
-                detector.writeTo(outputPath);
+        // Currently there could two runners, one for JUnit 4 and one for JUnit 5
+        // If the maven project has both JUnit 4 and JUnit 5 tests, two runners will
+        // be returned
+        List<Runner> runners = RunnerFactory.allFrom(project);
+        runners = removeZombieRunners(runners, project);
+
+        if (runners.size() != 1) {
+            String errorMsg;
+            if (runners.size() == 0) {
+                errorMsg =
+                    "Module is not using a supported test framework (probably not JUnit), " +
+                    "or there is no test.";
             } else {
-                final String errorMsg = "Module has no tests, not running detector.";
-                TestPluginPlugin.info(errorMsg);
-                logger.writeError(errorMsg);
+                // more than one runner, currently is not supported.
```

Comment:
Did iDFlakies/testrunner support projects with both JUnit 4 and JUnit 5 tests before these changes? If so, can you add the gradle functionality and still have it support such projects?

Reply:
Well, it is due to merge error. Now I do not do anything to these statements.